pkg/controller/kieapp/defaults/auth.go

```diff
@@ -0,0 +1,93 @@
+package defaults
+
+import (
+	"fmt"
+
+	"github.com/kiegroup/kie-cloud-operator/pkg/apis/app/v1"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/constants"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/shared"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const ssoHostnameVar = "HOSTNAME_HTTPS"
+const ssoClientVar = "SSO_CLIENT"
+
+// ConfigureHostname sets the HOSTNAME_HTTPS environment variable with the provided hostname
+// IF not yet set AND SSO auth is configured AND SSO_CLIENT exists
+func ConfigureHostname(object *v1.CustomObject, cr *v1.KieApp, hostname string) {
+	if cr.Spec.Auth.Type != v1.SSOAuthType {
+		return
+	}
+	for dcIdx := range object.DeploymentConfigs {
+		dc := &object.DeploymentConfigs[dcIdx]
+		for containerIdx := range dc.Spec.Template.Spec.Containers {
+			container := &dc.Spec.Template.Spec.Containers[containerIdx]
+			if pos := shared.GetEnvVar(ssoClientVar, container.Env); pos == -1 {
+				continue
+			}
+			if pos := shared.GetEnvVar(ssoHostnameVar, container.Env); pos == -1 {
+				container.Env = append(container.Env, corev1.EnvVar{
+					Name:  ssoHostnameVar,
+					Value: hostname,
+				})
+			} else if len(container.Env[pos].Value) == 0 {
+				container.Env[pos].Value = hostname
+			}
+		}
+	}
+}
+
+func configureAuth(spec v1.KieAppSpec, envTemplate *v1.EnvTemplate) error {
+	var err error
+	if len(spec.Auth.Type) == 0 {
+		return nil
+	}
+	if spec.Auth.Type == v1.SSOAuthType {
+		err = configureSSO(spec.Auth.SSO, spec.KieDeployments, envTemplate)
+	} else if spec.Auth.Type == v1.LDAPAuthType {
+		err = configureLDAP(spec.Auth.LDAP, envTemplate)
+	} else {
+		return fmt.Errorf("invalid authentication type %v", spec.Auth.Type)
+	}
+	if spec.Auth.RoleMapper != nil {
+		err = configureRoleMapper(spec.Auth.RoleMapper, envTemplate)
+	}
+	return err
+}
```

Comment:
How about this, specially for this case where you have a lot of ifs, and declare an error upfront:

```
func configureAuth(spec v1.KieAppSpec, envTemplate *v1.EnvTemplate) (err error) {
	switch {
	case len(spec.Auth.Type) == 0:
		err = nil
	case spec.Auth.Type == v1.SSOAuthType:
		err = configureSSO(spec.Auth.SSO, spec.KieDeployments, envTemplate)
	case spec.Auth.Type == v1.LDAPAuthType:
		err = configureLDAP(spec.Auth.LDAP, envTemplate)
	default:
		err = fmt.Errorf("invalid authentication type %v", spec.Auth.Type)
	}

	if err == nil && spec.Auth.RoleMapper != nil {
		err = configureRoleMapper(spec.Auth.RoleMapper, envTemplate)
	}
	return
}
```

Reply:
After removing the type I don't think see how to avoid the use of the ifs. I have however declared the error as suggested.

Reply:
FWIW, in Go, you can use switch/case in the format we usually have to use if/else in Java. It's not necessarily or always better, but can something be cleaner. That's the case for my example above, as I'm not "switching on anything". You can in fact still do so, though it doesn't change much.